bundles/org.eclipse.swt/Eclipse SWT/cocoa/org/eclipse/swt/widgets/Display.java

```diff
@@ -2502,6 +2506,26 @@ public void run() {
 	isPainting = isPainting.initWithCapacity(12);
 }
 
+/**
+ * Checks if the native item height should be enforced as a minimum.
+ * 
+ * Newer version of macOS may use a default item height in Table, Tree and List
+ * controls that is larger than what is traditionally expected.
+ * 
+ * Enforcing the default height as a minimum may break existing assumptions and
+ * render UI elements with a padding that may be considered too large.
+ * 
+ * This is implicitly enabled unless {@code org.eclipse.swt.internal.carbon.smallFonts} is set.
+ */
+private boolean initDefaultEnforceNativeItemHeightMinimum() {
+	boolean enforce = !Device.SMALL_FONTS;
+	String v = System.getProperty("org.eclipse.swt.internal.cocoa.enforceNativeItemHeightMinimum");
+	if (v != null) {
+		enforce = "true".equals(v);
+	}
+	return enforce;
+}
+
```

Comment:
Users might want small fonts _and_ native item height. See https://github.com/eclipse-platform/eclipse.platform.ui/issues/1674#issuecomment-1956169651

Reply:
see my comment below — the native height is smaller since we use `NSFont.smallSystemFontSize()`, which is taken into account.